### PR TITLE
Monitoring, fetch browser data. Send to server. Show local link info …

### PR DESCRIPTION
**Pull request by bbfsdev**
…(only with deb param set).

### CODE REVIEW

File: src/apps/VirtualApp/components/Monitoring.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  Icon,
+  Menu,
+  Placeholder,
+  Popup,
+  Table,
+} from 'semantic-ui-react';
+
+const usePrevious = (value) => {
+  const ref = useRef();
+  useEffect(() => {
```
Comment (gshilin) [inline src/apps/VirtualApp/components/Monitoring.js:16]: What do you need useEffect for?

File: src/apps/VirtualApp/components/MonitoringData.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Monitoring library to track connection stats.
+import {
+  MONITORING_BACKEND,
+} from "../../../shared/consts";
+
+const ONE_SECOND_IN_MS = 1000;
+const STORE_INTERVAL = 60 * ONE_SECOND_IN_MS; // Store for one minute.
+
+export const MonitoringData = class {
```
Comment (gshilin) [inline src/apps/VirtualApp/components/MonitoringData.js:9]: Why not regular function?

File: src/apps/VirtualApp/components/Monitoring.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export const Monitoring = (props) => {
+  const {monitoringData} = props;
+	const [data, setData] = useState([]);
```
Comment (gshilin) [inline src/apps/VirtualApp/components/Monitoring.js:24]: improper indentation

File: src/apps/VirtualApp/components/Monitoring.js (hunk 1 above)
Comment (bbfsdev) [reply src/apps/VirtualApp/components/Monitoring.js:16]: This hook is just to get the previous value of a prop it is being used for.
See line 26

File: src/apps/VirtualApp/components/MonitoringData.js (hunk 2 above)
Comment (bbfsdev) [reply src/apps/VirtualApp/components/MonitoringData.js:9]: This is a library class, not React component, nor hook. Just simple javascript library.

File: src/apps/VirtualApp/components/Monitoring.js (hunk 3 above)
Comment (bbfsdev) [reply src/apps/VirtualApp/components/Monitoring.js:24]: Fixed.

File: src/apps/VirtualApp/components/MonitoringData.js (hunk 2 above)
Comment (gshilin) [reply src/apps/VirtualApp/components/MonitoringData.js:9]: Still class in not a native datatype in javascript, it's compiled to a function anyway, but to quite a complex one.

Comment (bbfsdev) [reply src/apps/VirtualApp/components/MonitoringData.js:9]: Yes, you are correct. This is more of a style guide. Please let me know what is preferred.

Comment (gshilin) [reply src/apps/VirtualApp/components/MonitoringData.js:9]: That's called Prototypal OO and it's supported directly even by IE6 ;)

A Class in JS:
function Animal(){  
    // Private property
    var alive=true;
    // Private method
    function fight(){ //... }   
    // Public method which can access private variables
    this.isAlive = function() { return alive; } 

    // Public property
    this.name = "Joe";
}

// Public method
Animal.prototype.play = function() { alert("Bow wow!"); }
// .. and so on

Now when you create it's object

var obj = new Animal();

**Comment by amnonbb** [comment]
Please pull last changes from master. First user enter flow is changed. We now need to fetch room list first. Only after that we know to which Janus server we going to connect. And so if user want to change room we have to reconnect.

**Pull request closed without merging**